code/scripting/api/objs/hudconfig.cpp

```diff
@@ -0,0 +1,237 @@
+#include "hudconfig.h"
+
+#include "hud/hudconfig.h"
+#include "ui/ui.h"
+#include "scripting/api/objs/color.h"
+#include "iff_defs/iff_defs.h"
+#include "globalincs/alphacolors.h"
+
+namespace scripting {
+namespace api {
+
+gauge_config_h::gauge_config_h() : gauge(-1) {}
+gauge_config_h::gauge_config_h(int l_gauge) : gauge(l_gauge) {}
+
+HC_gauge_region* gauge_config_h::getGauge() const
+{
+	return &HC_gauge_regions[GR_1024][gauge];
+}
+
+int gauge_config_h::getIndex() const
+{
+	return gauge;
+}
+
+const char* gauge_config_h::getName() const
+{
+	return HC_gauge_descriptions(gauge);
+}
+
+hud_preset_h::hud_preset_h() : preset(-1) {}
+hud_preset_h::hud_preset_h(int l_preset) : preset(l_preset) {}
+
+int hud_preset_h::getIndex() const
+{
+	return preset;
+}
+
+SCP_string hud_preset_h::getName() const
+{
+	return HC_preset_filenames[preset];
+}
+
+//**********HANDLE: hud preset
+ADE_OBJ(l_HUD_Preset, hud_preset_h, "hud_preset", "Hud preset handle");
+
+ADE_VIRTVAR(Name, l_HUD_Preset, nullptr, "The name of this preset", "string", "The name")
```

Comment:
The API functions in this file should check that `current` is valid before using it.  See also #5326.